### DISCUSSION TITLE
Request: A: Can we let Copilot vim plugin work behind proxy? B: My Linux server only get terminal environment, how can I finish setup steps?

### DISCUSSION THREAD

**Discussion by CodingPapi** [community/community#7017] [Copilot Conversations] [answered]
A: Can we let vim plugin work behind proxy? 
B: My Linux server only get terminal environment, how can I finish setup steps?

**Comment by tpope** [comment, marked as answer]
There's no solution for proxies yet. As for setting up in a terminal environment, it should work just fine. You'll just need to open the URLs it shows manually.

**Reply by CodingPapi** [reply]
It will be greate to add proxy support, my server stuck at doing httprequest. Thanks for your hard work to make copilot  work on vim

**Reply by kimusan** [reply]
> There's no solution for proxies yet. As for setting up in a terminal environment, it should work just fine. You'll just need to open the URLs it shows manually.

That does not seem to work. I have opened the browser manually and authenticated and that part seems to work. This is however not fed back to the vim plugin and hence it will eventually tell me that it failed to open the browser. At this point :Copilot status still tell me that I am unauthenticated

**Reply by tpope** [reply]
The failure to open browser message is purely informative; it shouldn't prevent it from working. How long is "eventually"?

**Comment by Johannesklint** [comment]
Would love to have copilot working when on VPN. 
I have tried adding `.github.com` as a `NO_PROXY`in my `~/.zprofile` (where I have other proxy config for my work). It works fine for vscode, but that doesn't really do much for me, since I love working with nvim. 

@tpope Are there any plans on trying to make this work? 
Thanks for all the great work you do! 🙏

**Reply by tpope** [reply]
Proxy support has since been added and is documented [here](https://github.com/github/copilot.vim/blob/1358e8e45ecedc53daf971924a0541ddf6224faf/doc/copilot.txt#L77-L91). It doesn't support environment variables like `NO_PROXY`, but since "no proxy" is the default, I wouldn't expect that to matter in this case.
